test/fixtures/poro.rb

```diff
@@ -103,143 +111,135 @@ def blog
-  attributes :id, :name
-
-  has_many :locations
+class Bio < Model
+  attributes :content, :rating
+  associations :author
 end
-
 class BioSerializer < ActiveModel::Serializer
   cache except: [:content], skip_digest: true
   attributes :id, :content, :rating
 
   belongs_to :author
 end
 
+class Blog < Model
+  attributes :name, :type, :special_attribute
+  associations :writer, :articles
+end
 class BlogSerializer < ActiveModel::Serializer
   cache key: 'blog'
   attributes :id, :name
 
   belongs_to :writer
   has_many :articles
 end
-
-class PaginatedSerializer < ActiveModel::Serializer::CollectionSerializer
-  def json_key
-    'paginated'
-  end
-end
-
 class AlternateBlogSerializer < ActiveModel::Serializer
   attribute :id
   attribute :name, key: :title
 end
-
 class CustomBlogSerializer < ActiveModel::Serializer
   attribute :id
   attribute :special_attribute
-
   has_many :articles
 end
 
-class CommentPreviewSerializer < ActiveModel::Serializer
-  attributes :id
-
-  belongs_to :post
+class Role < Model
+  attributes :name, :description, :special_attribute
+  associations :author
 end
+class RoleSerializer < ActiveModel::Serializer
+  cache only: [:name, :slug], skip_digest: true
+  attributes :id, :name, :description
+  attribute :friendly_id, key: :slug
+  belongs_to :author
 
-class AuthorPreviewSerializer < ActiveModel::Serializer
-  attributes :id
+  def friendly_id
+    "#{object.name}-#{object.id}"
+  end
+end
 
-  has_many :posts
+class Location < Model
+  attributes :lat, :lng
+  associations :place
 end
+class LocationSerializer < ActiveModel::Serializer
+  cache only: [:address], skip_digest: true
+  attributes :id, :lat, :lng
 
-class PostPreviewSerializer < ActiveModel::Serializer
-  attributes :title, :body, :id
+  belongs_to :place, key: :address
 
-  has_many :comments, serializer: CommentPreviewSerializer
-  belongs_to :author, serializer: AuthorPreviewSerializer
+  def place
+    'Nowhere'
+  end
 end
 
-class PostWithTagsSerializer < ActiveModel::Serializer
-  attributes :id
+class Place < Model
+  attributes :name
+  associations :locations
+end
+class PlaceSerializer < ActiveModel::Serializer
+  attributes :id, :name
+  has_many :locations
+end
 
-  has_many :tags
+class Like < Model
+  attributes :time
+  attributes :likeable
+end
+class LikeSerializer < ActiveModel::Serializer
+  attributes :id, :time
+  belongs_to :likeable
 end
 
-class PostWithCustomKeysSerializer < ActiveModel::Serializer
-  attributes :id
+module Spam
+  class UnrelatedLink < Model
+  end
+  class UnrelatedLinkSerializer < ActiveModel::Serializer
+    cache only: [:id]
+    attributes :id
+  end
+end
 
-  has_many :comments, key: :reviews
-  belongs_to :author, key: :writer
-  has_one :blog, key: :site
+class Tag < Model
+  attributes :name
 end
 
+class VirtualValue < Model; end
 class VirtualValueSerializer < ActiveModel::Serializer
   attributes :id
-
   has_many :reviews, virtual_value: [{ type: 'reviews', id: '1' },
```

Comment:
I think we should remove it. It's not as clean as defining a reviews method, or using the block form.